### ISSUE TITLE
Implement `print` method for S3 objects

### ISSUE THREAD

**Issue by bundfussr** [pharmaverse/admiral#858]
For all S3 objects provided by admiral a `print` method should be provided. This could be implemented be using a superclass like `source` and implementing the method for `source` only. See `print.tte_source()` for an example.

**Comment by bms63** [comment]
@bundfussr I'm unsure which objects within admiral need this print method implemented.  

Does each object need similar print output as done in `print.tte.soure.`  Or it will be determined by what the object is.  


```
print.tte_source <- function(x, ...) {
  cat <- function(...) base::cat(..., sep = "")
  cat("<new_source> object\n")    # New Source
  cat("dataset_name: \"", x$dataset_name, "\"\n")
  cat("filter: ", quo_text(x$filter), "\n")
  cat("date: ", quo_text(x$date), "\n")
  cat("censor: ", x$censor, "\n")  # Behavior of object
  cat("set_values_to:\n")
  for (name in names(x$set_values_to)) {
    cat("  ", name, ": ", quo_text(x$set_values_to[[name]]), "\n")
  }
}
```

**Comment by bms63** [comment]
Also, if we have several of these print methods.  Do we want to have a standard convention like they do for purrr and rlang pirint methods.   I used methods(print) to get these.  

![image](https://user-images.githubusercontent.com/10111024/154861136-089931ea-7709-4b29-a27b-bbd76a1bc59e.png)


For us maybe just `print.admiral_tte_source`, `print_admiral_new_thing1`, `print_admiral_new_thing2`

**Comment by bundfussr** [comment]
We should have a `print` method for all classes in the ["Source Specifications" section](https://pharmaverse.github.io/admiral/reference/index.html#section-source-specifications). This does not require a separate function for each class. E.g., `print_tte_source()` is used for printing both `censor_source` and `event_source`.

Regarding names ["Advanced R"](https://adv-r.hadley.nz/s3.html#s3-classes) recommends adding the package name to the class name.

@thomas-neitmann , what do you think? Should we follow this recommendation?

**Comment by thomas-neitmann** [comment]
I don't have strong feelings reagrding the naming. It certainly avoids name clashes with classes from other packages when prefixing the class name with the package name but it also becomes very verbose. I'll leave it up to you to decide.

Regarding implementation, if we make all the source object inherits from a common `source` class then we can get away with writing a single `print.source()` function. I would imagine the output of all these objects to look similar to the existing `print.tte_source()` one. Basically,

```
<object_name> object
<name>: <value>
<name>: <value>
...
```

**Comment by bms63** [comment]
Should I name it` print.obj_source()` so it is generic and captures all four of the objects?  We can leave off the `admiral` unless you all think there will be more class stuff like this.     

There is a `params` object as well.  Is that one in scope for this update?

**Comment by thomas-neitmann** [comment]
> Should I name it print.obj_source() so it is generic and captures all four of the objects?

Yes, something along those lines. Just make sure that in the constructor functions for these object the class is set appropriatly.

> There is a params object as well. Is that one in scope for this update?

Not quite sure. Seems a bit different from the others, though not too much. I^ll leave that to your judgment.

**Comment by bms63** [comment]
@thomas-neitmann  @bundfussr  @duniek 

Could this one be a good hackR item?  I haven't been able to get back to it.

**Comment by Natnassonova** [comment]
Would it be possible to get assigned to this project?

**Comment by thomas-neitmann** [comment]
Sure :)

**Comment by rossfarrugia** [comment]
@Natnassonova are you still interested in contributing to this issue?

**Comment by Natnassonova** [comment]
@rossfarrugia I would like to stop contributing for now if that's ok

**Comment by rossfarrugia** [comment]
no problem, thanks for confirming! i unassigned you and put this back to the backlog for someone else to pick up later
